File: guilt/constants/paths.py
from pathlib import Path
from typing import Final

GUILT_DIRECTORY: Final = Path.home()

CPU_PROFILES_PATH: Final = GUILT_DIRECTORY / "cpu_profiles.json"
PROCESSED_JOBS_PATH: Final = GUILT_DIRECTORY / "processed_jobs.json"
UNPROCESSED_JOBS_PATH: Final = GUILT_DIRECTORY / "unprocessed_jobs.json"

File: src/guilt_directory/paths.rs
use crate::users::User;
use std::path::PathBuf;

pub const CPU_PROFILES_FILE: &str = "cpu_profiles.json";
pub const UNPROCESSED_JOBS_FILE: &str = "unprocessed_jobs.json";
pub const PROCESSED_JOBS_FILE: &str = "processed_jobs.json";
pub const LAST_WRITTEN_VERSION_FILE: &str = "last_written_version";
pub const PROFILE_RESOLUTION_POLICY_FILE: &str = "profile_resolution_policy.rhai";

pub const GUILT_DIR: &str = ".guilt";

pub fn guilt_directory_for_user(user: &User) -> PathBuf {
    let home_dir = user.home_dir.as_path();
    home_dir.join(GUILT_DIR)
}
